public/app/features/alerting/unified/RuleEditorCloudOnlyAllowed.test.tsx

```diff
@@ -0,0 +1,236 @@
+        <Route path={['/alerting/new', '/alerting/:id/edit']} component={RuleEditor} />
+      </Router>
+    </Provider>
+  );
+}
+
+const ui = {
+  inputs: {
+    name: byRole('textbox', { name: /rule name name for the alert rule\./i }),
+    alertType: byTestId('alert-type-picker'),
+    dataSource: byTestId('datasource-picker'),
+    folder: byTestId('folder-picker'),
+    folderContainer: byTestId(selectors.components.FolderPicker.containerV2),
+    namespace: byTestId('namespace-picker'),
+    group: byTestId('group-picker'),
+    annotationKey: (idx: number) => byTestId(`annotation-key-${idx}`),
+    annotationValue: (idx: number) => byTestId(`annotation-value-${idx}`),
+    labelKey: (idx: number) => byTestId(`label-key-${idx}`),
+    labelValue: (idx: number) => byTestId(`label-value-${idx}`),
+    expr: byTestId('expr'),
+  },
+  buttons: {
+    save: byRole('button', { name: 'Save' }),
+    addAnnotation: byRole('button', { name: /Add info/ }),
+    addLabel: byRole('button', { name: /Add label/ }),
+    // alert type buttons
+    lotexAlert: byRole('button', { name: /Mimir or Loki alert/ }),
+  },
+};
+
+describe('RuleEditor cloud: checking editable data sources', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    contextSrv.isEditor = true;
+    contextSrv.hasEditPermissionInFolders = true;
+  });
+
+  disableRBAC();
+
+  it('for cloud alerts, should only allow to select editable rules sources', async () => {
+    const dataSources = {
+      // can edit rules
+      loki: mockDataSource(
+        {
+          type: DataSourceType.Loki,
+          name: 'loki with ruler',
+        },
+        { alerting: true }
+      ),
+      loki_disabled: mockDataSource(
+        {
+          type: DataSourceType.Loki,
+          name: 'loki disabled for alerting',
+          jsonData: {
+            manageAlerts: false,
+          },
+        },
+        { alerting: true }
+      ),
+      // can edit rules
+      prom: mockDataSource(
+        {
+          type: DataSourceType.Prometheus,
+          name: 'cortex with ruler',
+        },
+        { alerting: true }
+      ),
+      // cannot edit rules
+      loki_local_rule_store: mockDataSource(
+        {
+          type: DataSourceType.Loki,
+          name: 'loki with local rule store',
+        },
+        { alerting: true }
+      ),
+      // cannot edit rules
+      prom_no_ruler_api: mockDataSource(
+        {
+          type: DataSourceType.Loki,
+          name: 'cortex without ruler api',
+        },
+        { alerting: true }
+      ),
+      // not a supported datasource type
+      splunk: mockDataSource(
+        {
+          type: 'splunk',
+          name: 'splunk',
+        },
+        { alerting: true }
+      ),
+    };
+
+    mocks.api.discoverFeatures.mockImplementation(async (dataSourceName) => {
+      if (dataSourceName === 'loki with ruler' || dataSourceName === 'cortex with ruler') {
+        return {
```

Comment:
Let's try to add `mockDiscoverFeaturesResponse` like to get rid of these duplications
```ts
function mockDiscoverFeaturesReponse(application: PromApplication, features: Partial<PromApiFeatures['features']> {...}
```

Reply:
🙌 good idea!